### PR TITLE
Support simple Hadoop impersonation

### PR DESCRIPTION
**Pull request by drdee**
@yagnik @dterror

### CODE REVIEW

**Comment by dterror-zz** [comment]
:+1:

File: test/test_viking_instantiation.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require File.expand_path('../../lib/viking.rb', __FILE__)
+
+java_import org.apache.hadoop.security.UserGroupInformation
+
+class TestFileUtilsClassMethods < MiniTest::Unit::TestCase
+	def test_no_impersonation
```
Comment (yagnik) [inline test/test_viking_instantiation.rb:8]: indentation

File: lib/viking.rb [hunk 2]
```diff
@@ -5,25 +5,34 @@
...
   DEFAULT_BUFFER_SIZE = 4096
 
+  def self.set_username(conf)
+    username     = Java::java.lang.System.getProperty("HADOOP_USER_NAME") || UserGroupInformation.getCurrentUser().getShortUserName()
+    conf.set("hadoop.job.ugi", username);
+  end
+
   def self.configure(config)
-    path = URI.new(config[:path])
+    path         = URI.new(config[:path])
```
Comment (yagnik) [inline lib/viking.rb:23]: what's with ridiculous space ?

**Comment by yagnik** [comment]
what was the reason for this ?

**Pull request merged**